CMakeLists.txt

```diff
@@ -54,7 +57,7 @@ AUX_SOURCE_DIRECTORY(trantor/net/inner DIR_SRC)
 find_package (OpenSSL)
 if(OpenSSL_FOUND)
     #add_definitions(-DUSE_OPENSSL)
-    set(DEFS "USE_OPENSSL;${DEFS}")
+    set(definitions "USE_OPENSSL;${definitions}")
     include_directories(${OPENSSL_INCLUDE_DIR})
     link_libraries(${OPENSSL_LIBRARIES})
     AUX_SOURCE_DIRECTORY(trantor/net/ssl DIR_SRC)
```

Comment:
When I used `find_package(OpenSSL CONFIG)`, I got an error on both MacOS and Linux:
```
Could NOT find OpenSSL (missing: OpenSSL_DIR)
``` 
And if I remove the `USE_OPENSSL` macro, How do I handle conditional compilation like this？
https://github.com/an-tao/trantor/blob/ad24f2c6733681f69c0c5bd9f218ef35c4f346d2/trantor/net/TcpServer.cc#L172-L193

Reply:
> When I used find_package(OpenSSL CONFIG), I got an error on both MacOS and Linux:

Oh, that's an old style FindCMake file...  o.O

In that case it requires four lines instead of one for optional inclusion.  ^.^

```cmake
find_package(OpenSSL REQUIRED)
if(OpenSSL_FOUND)
    target_link_libraries(whatever_the_target_name_is PUBLIC OpenSSL::SSL OpenSSL::Crypto)
endif()
```
The `target_link_libraries` command will auto-link in the public headers, libraries, etc...  No need for the multiple manual commands.  :-)

> And if I remove the USE_OPENSSL macro, How do I handle conditional compilation like this？

A `USE_OPENSSL` macro would be appropriate here, but as it changes the API of the library this is one that you will want to bake into the INSTALL scripts as well via a template config.h.in or so.

Reply:
I removed USE_OPENSSL macro from API header files now, this problem may be resolved.